addons/account_bank_statement_import/account_bank_statement_import.py

```diff
@@ -2,64 +2,54 @@
 
 import base64
 
-from openerp import SUPERUSER_ID
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
+from openerp import models, fields, api, _
 from openerp.exceptions import UserError
 
 import logging
 _logger = logging.getLogger(__name__)
 
 
-class account_bank_statement_line(osv.osv):
+class account_bank_statement_line(models.Model):
     _inherit = "account.bank.statement.line"
 
-    _columns = {
-        # Ensure transactions can be imported only once (if the import format provides unique transaction ids)
-        'unique_import_id': fields.char('Import ID', readonly=True, copy=False),
-    }
+    # Ensure transactions can be imported only once (if the import format provides unique transaction ids)
+    unique_import_id = fields.Char(string='Import ID', readonly=True, copy=False)
 
     _sql_constraints = [
         ('unique_import_id', 'unique (unique_import_id)', 'A bank account transactions can be imported only once !')
     ]
 
 
-class account_bank_statement_import(osv.TransientModel):
+class account_bank_statement_import(models.TransientModel):
     _name = 'account.bank.statement.import'
     _description = 'Import Bank Statement'
-    _columns = {
-        'data_file': fields.binary('Bank Statement File', required=True, help='Get you bank statements in electronic format from your bank and select them here.'),
-    }
 
-    def import_file(self, cr, uid, ids, context=None):
+    data_file = fields.Binary(string='Bank Statement File', required=True, help='Get you bank statements in electronic format from your bank and select them here.')
+
+    @api.multi
+    def import_file(self):
         """ Process the file chosen in the wizard, create bank statement(s) and go to reconciliation. """
-        if context is None:
-            context = {}
         #set the active_id in the context, so that any extension module could
         #reuse the fields chosen in the wizard if needed (see .QIF for example)
-        context.update({'active_id': ids[0]})
-
-        data_file = self.browse(cr, uid, ids[0], context=context).data_file
-
+        self = self.with_context(active_id=self.ids[0])
```

Comment:
Reassigning 'self' in a method is probably unnecessary and it may break something !!!